src/server/common/helpers/auth/fetch.js

```diff
@@ -0,0 +1,100 @@
+import Boom from '@hapi/boom'
+import fetch from 'node-fetch'
+import { getUserSession } from '#server/common/helpers/auth/get-user-session.js'
+
+export async function fetchWithAuthHeader(url, request) {
+  const accessToken = (await getUserSession(request)).token
+  const response = await fetch(url, {
+    headers: {
+      Authorization: `Bearer ${accessToken}`
+    }
+  })
+
+  const data = await response.json()
+
+  return { response, data }
+}
+
+function getOrganisationLinkHref(id, redirectUrl) {
+  return `/organisations/${id}/link?redirectUrl=${redirectUrl}`
+}
+
+export async function fetchWithAuthInterception(url, request, h) {
+  request.logger.info({ url }, 'auth: fetchWithAuthInterception')
+
+  try {
+    const { data, response } = await fetchWithAuthHeader(url, request)
+
+    if (!response.ok) {
+      throw Boom.unauthorized()
+    }
+
+    if (data.action === 'link-organisations') {
+      const { path: redirectUrl } = request
+      const { defraId, isCurrentOrganisationLinked, organisations } = data
+      const hasManyOrganisations = organisations.length > 1
+      const entityName = hasManyOrganisations ? 'organisations' : 'organisation'
+      const shouldShowTableOfUnlinkedOrganisations =
+        hasManyOrganisations && !isCurrentOrganisationLinked
+      const otherRelationships = defraId.otherRelationships.map(
+        ({ defraIdOrgName, defraIdRelationshipId }) => [
+          { text: defraIdOrgName },
+          {
+            html: `<a href="http://localhost:3200/cdp-defra-id-stub/register/${defraId.userId}/relationship/${defraIdRelationshipId}/current">Switch</a>`
+          }
+        ]
+      )
+
+      const commonTemplateData = {
+        defraIdOrgName: defraId.orgName,
+        entityName,
+        isCurrentOrganisationLinked,
+        otherRelationships,
+        shouldShowTableOfUnlinkedOrganisations
+      }
+
+      const templateData = shouldShowTableOfUnlinkedOrganisations
+        ? {
+            ...commonTemplateData,
+            organisations: organisations.map(({ name, orgId, id }) => [
+              { text: name ?? 'data missing' },
+              { text: orgId ?? 'data missing' },
+              { text: id ?? 'data missing' },
```

Comment:
I don't think this defensive coding approach is necessary as I've never seen any of these fields actually missing values